autoreject/autoreject.py

```diff
@@ -469,34 +454,32 @@ def _vote_bad_epochs(self, epochs):
             The epochs object for which bad epochs must be found.
         """
         n_epochs = len(epochs)
-        picks = _handle_picks(info=epochs.info, picks=self.picks)
+        picks = _handle_picks(info=epochs.info, picks=picks)
 
-        drop_log = np.zeros((n_epochs, len(epochs.ch_names)))
+        ch_names = [cc for cc in epochs.ch_names]
+        drop_log = np.zeros((n_epochs, len(ch_names)))
         bad_sensor_counts = np.zeros((len(epochs), ))
 
-        ch_names = [epochs.ch_names[p] for p in picks]
+        this_ch_names = [epochs.ch_names[p] for p in picks]
         deltas = np.ptp(epochs.get_data()[:, picks], axis=-1).T
-        threshes = [self.threshes_[ch_name] for ch_name in ch_names]
+        threshes = [self.threshes_[ch_name] for ch_name in this_ch_names]
         for ch_idx, (delta, thresh) in enumerate(zip(deltas, threshes)):
             bad_epochs_idx = np.where(delta > thresh)[0]
             # TODO: combine for different ch types
             bad_sensor_counts[bad_epochs_idx] += 1
             drop_log[bad_epochs_idx, picks[ch_idx]] = 1
-        return drop_log, bad_sensor_counts
+        return drop_log, bad_sensor_counts, ch_names
 
     def _get_epochs_interpolation(self, epochs, drop_log,
-                                  ch_type, verbose='progressbar'):
+                                  ch_type, picks, n_interpolate,
+                                  verbose='progressbar'):
         """Interpolate the bad epochs."""
         # 1: bad segment, # 2: interpolated
-        fix_log = drop_log.copy()
-        ch_names = epochs.ch_names
-        non_picks = np.setdiff1d(range(epochs.info['nchan']), self.picks)
-        interp_channels = list()
-        n_interpolate = self.n_interpolate[ch_type]
+        labels = drop_log.copy()
+        non_picks = np.setdiff1d(range(epochs.info['nchan']), picks)
         for epoch_idx in range(len(epochs)):
-            n_bads = drop_log[epoch_idx, self.picks].sum()
+            n_bads = drop_log[epoch_idx, picks].sum()
             if n_bads == 0:
-                interp_channels.append([])
```

Comment:
this needs to be fixed

Reply:
What exactly? It is obsolete now. We extract the bad interp chans from fix log.

Reply:
oh I see. Makes sense